[tests] Use ResiliencePipeline in IntegrationTests

radical opened the pull request:
.. with a policy to retry with 1 sec delays, and a total timeout of
90 secs on the service side. The test(client) side has an overall http
request timeout for 120 secs which would allow the real errors from the
service to get surfaced to the client.

Fixes #3366

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3385)

radical committed:
[tests] Use ResiliencePipeline in IntegrationTests

.. with a policy to retry with 1 sec delays, and a total timeout of
90 secs on the service side. The test(client) side has an overall http
request timeout for 120 secs which would allow the real errors from the
service to get surfaced to the client.

radical commented on tests/testproject/TestProject.IntegrationServiceA/TestUtils.cs:
```diff
@@ -0,0 +1,24 @@
+
+namespace Aspire.TestProject;
+
+public static class TestUtils
```
better name?

eerhardt replied:
Maybe ResilienceUtils?

radical replied:
done.

radical commented on tests/testproject/TestProject.IntegrationServiceA/SqlServer/SqlServerExtensions.cs:
```diff
@@ -13,14 +16,16 @@ public static void MapSqlServerApi(this WebApplication app)
 
     private static async Task<IResult> VerifySqlServerAsync(SqlConnection connection)
     {
+        StringBuilder errorMessageBuilder = new();
```
In a follow up PR I will add listening to logs from the services in `AppHost`, so this shouldn't be needed then.

eerhardt commented on tests/testproject/TestProject.IntegrationServiceA/TestProject.IntegrationServiceA.csproj:
```diff
@@ -32,6 +32,7 @@
     <ProjectReference Condition="'$(TestsRunningOutsideOfRepo)' != 'true'" Include="@(ComponentReferenceForTests -> '$(RepoRoot)src\Components\%(Identity)\%(Identity).csproj')" />
     <PackageReference Condition="'$(TestsRunningOutsideOfRepo)' == 'true'" Include="@(ComponentReferenceForTests)" />
     <PackageReference Include="Polly" />
```
```suggestion
```

We can remove this now that we are using Http.Resilience.

Actually, we shouldn't be using Http.Resilience, that is for Http calls. We should just need to change this to `Polly.Core`.

radical commented on tests/testproject/TestProject.IntegrationServiceA/Cosmos/CosmosExtensions.cs:
```diff
@@ -13,15 +16,17 @@ public static void MapCosmosApi(this WebApplication app)
-                .WaitAndRetryAsync(60, retryAttempt => TimeSpan.FromSeconds(1));
+            ResiliencePipeline pipeline = TestUtils.GetDefaultResiliencePipelineBuilder<HttpRequestException>(args =>
+            {
+                errorMessageBuilder.AppendLine(CultureInfo.InvariantCulture, $"{Environment.NewLine}Service retry #{args.AttemptNumber} due to {args.Outcome.Exception}");
```
`error CA1305: The behavior of 'StringBuilder.AppendLine(ref StringBuilder.AppendInterpolatedStringHandler)' could vary based on the current user's locale settings. Replace this call in 'OracleDatabaseExtensions.VerifyOracleDatabase(MyDbContext)' with a call to 'StringBuilder.AppendLine(IFormatProvider, ref StringBuilder.AppendInterpolatedStringHandler)'. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305)`

eerhardt replied:
Can we just add

https://github.com/dotnet/aspire/blob/0a28bb902c8c5271252f3f76236f5c44f371839f/src/Tools/ConfigurationSchemaGenerator/RuntimeSource/.editorconfig#L6-L7

to 

https://github.com/dotnet/aspire/blob/0a28bb902c8c5271252f3f76236f5c44f371839f/tests/.editorconfig#L1-L4

?

radical replied:
done.

radical committed:
address review feedback from @ eerhardt

radical committed:
Use the pipeline for MySql also

radical committed:
IntegrationServiceFixture.DumpComponentLogsAsync: error out if more than one resource is given

radical committed:
Rename TestUtils -> ResilienceUtils - feedback from @ eerhardt

radical committed:
[tests] Ignore `CA1305` for tests

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1305

```
dotnet_diagnostic.CA1305.severity = none
```

Addresses review feedback from @ eerhardt .

radical committed:
Merge branch 'dotnet:main' into integ-tests-timeout

eerhardt committed:
Remove Polly reference

eerhardt commented on tests/testproject/TestProject.IntegrationServiceA/ResilienceUtils.cs:
```diff
@@ -0,0 +1,24 @@
+
+public static class ResilienceUtils
+{
+    public static ResiliencePipelineBuilder GetDefaultResiliencePipelineBuilder<TException>(Func<OnRetryArguments<object>, ValueTask> onRetry, int overallTimeoutSecs = 90) where TException : Exception
```
Can you add a comment explaining the strategy here?

eerhardt committed:
Make code consistent.

eerhardt reviewed (approved):
LGTM

radical committed:
add comment addressing review feedback from @ eerhardt

radical committed:
Merge remote-tracking branch 'origin/main' into integ-tests-timeout

# Conflicts:
#	Directory.Packages.props

radical commented:
/backport to release/8.0
